### DISCUSSION TITLE
Potential inconsistency between displacement height and reference height from ground

### DISCUSSION THREAD

**Discussion by hjgim** [NCAR/noahmp#159] [General]
Displacement height (ZeroPlaneDispSfc) for a vegetation tile is determined as 65% of the canopy top height using the following expression:

ZeroPlaneDispSfc = 0.65 * HeightCanopyTop
https://github.com/NCAR/noahmp/blob/fa2d2098bac1e437a21ccc15494b152996d5bc84/src/GroundRoughnessPropertyMod.F90#L63

The reference height from the ground (RefHeightAboveGrd) is then calculated using:

RefHeightAboveGrd = max(ZeroPlaneDispSfc, HeightCanopyTop) + RefHeightAboveSfc
https://github.com/NCAR/noahmp/blob/fa2d2098bac1e437a21ccc15494b152996d5bc84/src/GroundRoughnessPropertyMod.F90#L79

Here, since ZeroPlaneDispSfc is defined as 65% of HeightCanopyTop, the expression max(ZeroPlaneDispSfc, HeightCanopyTop) will always be equal to HeightCanopyTop when there is no snow layer. Thus, the above formula can be simplified as follows in snow-free conditions:

RefHeightAboveGrd = HeightCanopyTop + RefHeightAboveSfc

**Issue for Discussion**
As I understand it, if we presume the bottom end of the wind log-profile is at the displacement height, the above formula needs to be modified as follows:

RefHeightAboveGrd = ZeroPlaneDispSfc + RefHeightAboveSfc

Could you please review this potential inconsistency and proposed code modification, and share your feedback?

**Comment by cenlinhe** [comment]
@RAbolafiaRosenzweig @tslin2 @barlage Could you please take a look at this?

**Reply by tslin2** [reply]
I had the same question when I saw this code a few months ago.
I also think this may be revised to RefHeightAboveGrd = RefHeightAboveSfc if forcing height is above canopy

**Reply by tslin2** [reply]
> I did some tests for this in a small region in the southeast US in WRF, the 2m temperature difference is about 0.2-0.3 degree, but I haven't looked at the wind speed.

**Reply by cenlinhe** [reply]
which version did you test? this one: RefHeightAboveGrd = ZeroPlaneDispSfc + RefHeightAboveSfc, or RefHeightAboveGrd = RefHeightAboveSfc?

**Reply by tslin2** [reply]
I did both; these two difference in 2 m temperature are average about 0.1 degree

**Comment by cenlinhe** [comment]
@barlage @RAbolafiaRosenzweig What do you think?

**Comment by hjgim** [comment]
I tested for the globe with following modification:

RefHeightAboveGrd    = max(ZeroPlaneDispSfc, HeightCanopyTop) + RefHeightAboveSfc
to 
RefHeightAboveGrd = ZeroPlaneDispSfc + RefHeightAboveSfc

As the effects of the modification, as @tslin2 mentioned above, the 2-m temperature differences are about 0.2 K; the diurnal temperature range has become smaller. Daytime temperature reduces and night temperature increases.
Additionally, the 10-meter wind speed also shows differences, decreasing by approximately 0.2 m/s.

**Comment by hjgim** [comment]
This discussion is tracked in issue #172 and PR #207. For efficiency, this thread is now closed.
